pkg/storage/gc/gc.go

```diff
@@ -0,0 +1,687 @@
+package gc
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/rand"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/docker/distribution/registry/storage/driver"
+	godigest "github.com/opencontainers/go-digest"
+	ispec "github.com/opencontainers/image-spec/specs-go/v1"
+	oras "github.com/oras-project/artifacts-spec/specs-go/v1"
+
+	zerr "zotregistry.io/zot/errors"
+	zcommon "zotregistry.io/zot/pkg/common"
+	zlog "zotregistry.io/zot/pkg/log"
+	mTypes "zotregistry.io/zot/pkg/meta/types"
+	"zotregistry.io/zot/pkg/scheduler"
+	"zotregistry.io/zot/pkg/storage"
+	common "zotregistry.io/zot/pkg/storage/common"
+	"zotregistry.io/zot/pkg/storage/types"
+)
+
+const (
+	cosignSignatureTagSuffix = "sig"
+	SBOMTagSuffix            = "sbom"
+)
+
+type Options struct {
+	Referrers bool
+	// will garbage collect blobs older than
+	Delay time.Duration
+	// will garbage collect untagged manifests older than
+	RetentionDelay time.Duration
+}
+
+type GarbageCollect struct {
+	imgStore types.ImageStore
+	opts     Options
+	metaDB   mTypes.MetaDB
+	log      zlog.Logger
+}
+
+func NewGarbageCollect(imgStore types.ImageStore, metaDB mTypes.MetaDB, opts Options, log zlog.Logger,
+) GarbageCollect {
+	return GarbageCollect{
+		imgStore: imgStore,
+		metaDB:   metaDB,
+		opts:     opts,
+		log:      log,
+	}
+}
+
+func (gc GarbageCollect) RunPeriodically(interval time.Duration, sch *scheduler.Scheduler) {
+	generator := &GCTaskGenerator{
+		imgStore: gc.imgStore,
+		gc:       gc,
+	}
+
+	sch.SubmitGenerator(generator, interval, scheduler.MediumPriority)
+}
+
+func (gc GarbageCollect) Repo(repo string) error {
+	gc.log.Info().Msg(fmt.Sprintf("executing GC of orphaned blobs for %s", path.Join(gc.imgStore.RootDir(), repo)))
```

Comment:
I think there are too many non-debug logs. Maybe make them debug logs.

Reply:
I preserved these ones, and I put the rest on debug. These ones are checked in our tests.